cypress/parallel-features.sh

```diff
@@ -4,16 +4,23 @@
 function join_by { local IFS="$1"; shift; echo "$*"; }
 
 # Arguments:
-CUR_JOB=$1
+CUR_JOB=$(expr $1 - 1)
 MAX_JOBS=$2
 
 # Features
 FEATURE_LIST=( $(find cypress/integration/ -maxdepth 1 -name "*.feature") )
 
 # Calculation
-MAX_FEATURES=$(find cypress/integration/ -maxdepth 1 -name "*.feature" -printf '.' | wc -m)
-FEATURES_PER_JOB=$(expr $(expr ${MAX_FEATURES} + ${MAX_JOBS} - 1) / ${MAX_JOBS} )
-FEATURES_SKIP=$(expr $(expr ${CUR_JOB} - 1 ) \* ${FEATURES_PER_JOB} )
+MAX_FEATURES=$(find cypress/integration/ -maxdepth 1 -name "*.feature" -print | wc -l)
+# adds overhead features to the first jobs
+if [[ $CUR_JOB -lt $(expr ${MAX_FEATURES} % ${MAX_JOBS}) ]]
+then
+  FEATURES_PER_JOB=$(expr ${MAX_FEATURES} / ${MAX_JOBS} + 1)
+  FEATURES_SKIP=$(expr $(expr ${MAX_FEATURES} / ${MAX_JOBS} + 1) \* ${CUR_JOB})
+else
+  FEATURES_PER_JOB=$(expr ${MAX_FEATURES} / ${MAX_JOBS})
+  FEATURES_SKIP=$(expr $(expr ${MAX_FEATURES} / ${MAX_JOBS} + 1) \* $(expr ${MAX_FEATURES} % ${MAX_JOBS}) + $(expr $(expr ${MAX_FEATURES} / ${MAX_JOBS}) \* $(expr ${CUR_JOB} - ${MAX_FEATURES} % ${MAX_JOBS})))
```

Comment:
This part I do not understand, mind explaining?
Features: 20
Jobs: 8
Job: 5
Expected Skip: 4*9 = 36

(20 / 8 + 1 ) * (20 % 8) + ( (20 / 8) * ( 5 - 20%8)
(3) * (4) * ( (2) * (1) )
3 * 4 * 2 = 24
Actual Skip: 24

Besides this, I find the solution brilliant

Reply:
Following description in detail:


```bash
if [[ $CUR_JOB -lt $(expr ${MAX_FEATURES} % ${MAX_JOBS}) ]]
then
  FEATURES_PER_JOB=$(expr ${MAX_FEATURES} / ${MAX_JOBS} + 1)
  FEATURES_SKIP=$(expr $(expr ${MAX_FEATURES} / ${MAX_JOBS} + 1) \* ${CUR_JOB})
else
  FEATURES_PER_JOB=$(expr ${MAX_FEATURES} / ${MAX_JOBS})
  FEATURES_SKIP=$(expr $(expr ${MAX_FEATURES} / ${MAX_JOBS} + 1) \* $(expr ${MAX_FEATURES} % ${MAX_JOBS}) + $(expr $(expr ${MAX_FEATURES} / ${MAX_JOBS}) \* $(expr ${CUR_JOB} - ${MAX_FEATURES} % ${MAX_JOBS})))
fi
```

MAX_FEATURES: 17 (3 are named as *.feature.broken)
MAX_JOBS: 8


CUR_JOB 0:  (first CUR_JOB is Zero, because it’s starting at Zero until 7)

Is the only one that falls under the `then` at the moment:
  FEATURES_PER_JOB=(17 div 8 + 1) = 2 + 1 = 3
  FEATURES_SKIP=$(expr (17 div 8 + 1) * 0 = 0

CUR_JOB 1:

Falls under else:
  FEATURES_PER_JOB=(17 div 8) = 2
  FEATURES_SKIP=((17 div 8 + 1) * (17 mod 8) + ((17 div 8) * (1 - 17 mod 8)))
  = (3) * (1) + ((2) * (1 - 1)))
  = (3) * (1) + ((2) * (0)))
  = 3

CUR_JOB 2:

Falls under else:
  FEATURES_PER_JOB=(17 div 8) = 2
  FEATURES_SKIP=((17 div 8 + 1) * (17 mod 8) + ((17 div 8) * (2 - 17 mod 8)))
  = (3) * (1) + ((2) * (2 - 1)))
  = (3) * (1) + ((2) * (1)))
  = 5
  
…

CUR_JOB 7: (last one)

Falls under else:
  FEATURES_PER_JOB=(17 div 8) = 2
  FEATURES_SKIP=((17 div 8 + 1) * (17 mod 8) + ((17 div 8) * (7 - 17 mod 8)))
  = (3) * (1) + ((2) * (7 - 1)))
  = (3) * (1) + ((2) * (6)))
  = 15

Reply:
Ok I didn't get that it starts from 0 ;)

Reply:
```bash
CUR_JOB=$(expr $1 - 1)
```